extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClient.java

```diff
@@ -9,69 +9,195 @@
+    private static final String CALL_UNSUCCESSFUL_ERROR_TEMPLATE = "Call unsuccessful: %s";
+    private static final String TOKEN_LOOK_UP_SELF_PATH = "/v1/auth/token/lookup-self";
+    private static final String TOKEN_RENEW_SELF_PATH = "/v1/auth/token/renew-self";
     private static final int HTTP_CODE_404 = 404;
+    private static final String TIME_FORMAT = "%02dh:%02dm:%02ds";
+
     @NotNull
-    private final HashicorpVaultClientConfig hashicorpVaultConfig;
+    private final HashicorpVaultConfig hashicorpVaultConfig;
     @NotNull
     private final EdcHttpClient httpClient;
     @NotNull
+    private final ScheduledExecutorService scheduledExecutorService;
+    @NotNull
     private final ObjectMapper objectMapper;
+    @NotNull
+    private final Monitor monitor;
 
-    HashicorpVaultClient(@NotNull HashicorpVaultClientConfig hashicorpVaultConfig, @NotNull EdcHttpClient httpClient,
-                         @NotNull ObjectMapper objectMapper) {
+    HashicorpVaultClient(@NotNull HashicorpVaultConfig hashicorpVaultConfig,
+                         @NotNull EdcHttpClient httpClient,
+                         @NotNull ObjectMapper objectMapper,
+                         @NotNull ScheduledExecutorService scheduledExecutorService,
+                         @NotNull Monitor monitor) {
         this.hashicorpVaultConfig = hashicorpVaultConfig;
         this.httpClient = httpClient;
         this.objectMapper = objectMapper;
+        this.scheduledExecutorService = scheduledExecutorService;
+        this.monitor = monitor;
+    }
+
+    Result<HealthCheckResponse> doHealthCheck() {
+        var requestUri = getHealthCheckUrl();
+        var request = httpGet(requestUri);
+
+        try (var response = httpClient.execute(request)) {
+            var code = response.code();
+            var responseBody = Objects.requireNonNull(response.body()).string();
+            var responsePayload = objectMapper.readValue(responseBody, HealthCheckResponsePayload.class);
+            var healthCheckResponse = HealthCheckResponse.Builder
+                    .newInstance()
+                    .code(code)
+                    .payload(responsePayload)
+                    .build();
+            return Result.success(healthCheckResponse);
+        } catch (IOException e) {
+            return Result.failure(e.getMessage());
+        }
+    }
+
+    void inspectToken(TokenLookUpResponsePayloadToken token) {
+        if (token.isRootToken()) {
+            monitor.warning("Root token detected. Don't use root tokens in production environment!");
+        } else {
+            if (token.isRenewable()) {
+                monitor.info("Token is renewable");
+            } else {
+                monitor.warning("Token is not renewable");
+            }
+
+            if (token.isPeriodicToken()) {
+                monitor.info("Token has a renewal period of %s".formatted(token.getPeriod()));
```

Comment:
This does not make sense to log related information at two different levels. Please convert it to something that would be meaningful to an operator:

1. Is it a problem or something the operator must be made aware of? The fact that a token has a particular renewable period does not need to be reported. It pollutes the log stream.
2. If it is a problem, what is the specific remedy? Telling the operator to look at a property in a configuration file (for what?) does not indicate what they need to do. 
3. Do not report related information separately using different log levels.
4. Which token is it?